include/nbl/builtin/hlsl/prefix_sum_blur/box_sampler.hlsl

```diff
@@ -0,0 +1,183 @@
+        }
+        else
+        {
+            switch (wrapMode) {
+            case ETC_REPEAT:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T floored = prefixSumAccessor.template get<T, uint32_t>((lastIdx + leftFlIdx) % linearSize) + floor(T(leftFlIdx) / linearSize) * last;
+                const T ceiled = prefixSumAccessor.template get<T, uint32_t>((lastIdx + leftClIdx) % linearSize) + floor(T(leftClIdx) / linearSize) * last;
+                result -= lerp(floored, ceiled, alpha);
+                break;
+            }
+            case ETC_CLAMP_TO_BORDER:
+            {
+                result -= (leftIdx + 1) * borderColor;
+                break;
+            }
+            case ETC_CLAMP_TO_EDGE:
+            {
+                result -= (1 - abs(leftIdx)) * prefixSumAccessor.template get<T, uint32_t>(0);
+                break;
+            }
+            case ETC_MIRROR:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                T floored, ceiled;
+
+                if (abs(leftFlIdx + 1) % (2 * linearSize) == 0)
+                    floored = -(abs(leftFlIdx + 1) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftFlIdx + 1)) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        floored = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftFlIdx + 1) - 1) % linearSize);
+                    else
+                        floored = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>((leftFlIdx + 1) % linearSize - 1));
+                }
+
+                if (leftClIdx == 0) // Special case, wouldn't be possible for `floored` above
+                    ceiled = 0;
+                else if (abs(leftClIdx + 1) % (2 * linearSize) == 0)
+                    ceiled = -(abs(leftClIdx + 1) / linearSize) * last;
+                else
+                {
+                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftClIdx + 1)) / linearSize));
+                    if ((period & 0x1u) == 1)
+                        ceiled = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftClIdx + 1) - 1) % linearSize);
+                    else
+                        ceiled = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>((leftClIdx + 1) % linearSize - 1));
+                }
+
+                result -= lerp(floored, ceiled, alpha);
+                break;
+            }
+            case ETC_MIRROR_CLAMP_TO_EDGE:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T lastMinusOne = prefixSumAccessor.template get<T, uint32_t>(lastIdx - 1);
+                result -= (1 - abs(leftIdx)) * (last - lastMinusOne);

```

Comment:
you know left index is already negative, so make it `1+leftIdx` no need for the `abs`